timesketch/ui/explore/event.directive.ts

```diff
@@ -389,12 +408,73 @@ export const tsEvent = function () {
                 }
 
             })
+        },
+    }
+}
 
+export const tsEventAdd = ['$window', '$timeout', function ($window, $timeout) {
+    /**
+     * Render event details.
+     * @param sketch-id - The id for the sketch.
+     * @param meta - Metadata object returned from the datastore search.
+     * @param event - Event object.
+     */
+    return {
+        restrict: 'E',
+        template: require('./event-add.html'),
+        scope: {
+            sketchId: '=',
+            sketch: '=',
+            event: '=',
+            events: '=',
+            filter: '=',
+            query: '=',
+            queryDsl: '=',
+            index: '@',
+            addEventData: '=',
+        },
+        require: '^tsSearch',
+        controller: function ($scope, timesketchApi) {
+            $scope.clearAddEvent = function (event) {
+                $scope.addEventData[event._id].timestamp_desc = ''
+                $scope.addEventData[event._id].message = ''
+                $scope.addEventData[event._id].timestamp = event._source.datetime
+                $scope.addEventData[event._id].showForm = !$scope.addEvent[event._id].showForm
+            }
+            $scope.genEventId = function () {
+              let id = '';
+              const alpha = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_';
+              for (let i = 0; i < 20; i++) {
+                id += alpha.charAt(Math.floor(Math.random() * alpha.length));
+              }
+              return id;
+            }
         },
         link: function (scope, elem, attrs, ctrl) {
-            scope.getContext = function (event) {
```

Comment:
Removal of getContext could be the cause of broken Query Context functionnality.

Reply:
Good catch! Fixed in #619